feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,264 @@
+        self.variables = _find_all_variables(X, self.variables)
+        
+        # find categorical variables or check that those vars entered by the user
+        # are of type object
+        cat_variables = _find_categorical_variables(X[self.variables] if self.variables else X)
+        num_variables = _find_numerical_variables(X[self.variables] if self.variables else X)
+
+        data = pd.concat([X, y], axis=1) # column appending.
+        data.columns = list(X.columns) + ['target']
+
+        # split data into train and test sets
+        X_train, X_test, y_train, y_test = train_test_split(
+                                                data[cat_variables + num_variables + ['target']],
+                                                data['target'],
+                                                test_size = test_size,
+                                                random_state = random_state)
+
+        ## filling missing values.
+        for col in cat_variables:
+            X_train[col].fillna('M', inplace=True)
+            X_test[col].fillna('M', inplace=True)
+
+        for col in num_variables:
+            X_train[col].fillna(0, inplace=True)
+            X_test[col].fillna(0, inplace=True)
+
+        # categorical variables
+        if cat_variables: 
+            # create the mean encoder instance 
+            encoder = ce.MeanCategoricalEncoder()
+            # fit
+            encoder.fit(X_train[cat_variables], y_train)
+            # transform
+            X_train_cat_enc, X_test_cat_enc = encoder.transform(X_train[cat_variables]), encoder.transform(X_test[cat_variables])
+
+            # filling missing values on the results
+            X_train_cat_enc, X_test_cat_enc = X_train_cat_enc.fillna(0), X_test_cat_enc.fillna(0)
+
+        # numerical variables
+        self.numerical_binned_ = dict()
+        if num_variables: 
+            # Equal-frequency discretizer
+            disc = dsc.EqualFrequencyDiscretiser(q=quantiles, variables=num_variables, return_object=True)
+            # fit train data
+            disc.fit(X_train[num_variables])
+            # transform the data
+            X_train_t = disc.transform(X_train[num_variables])
+            X_test_t = disc.transform(X_test[num_variables])
+
+            # create the mean encoder instance 
+            encoder = ce.MeanCategoricalEncoder()
+            # mean encoding
+            encoder.fit(X_train_t[num_variables], y_train)
+            X_train_num_enc, X_test_num_enc = encoder.transform(X_train_t[num_variables]), encoder.transform(X_test_t[num_variables])
+            X_train_num_enc.columns = [i+'_binned' for i in X_train_num_enc.columns]
+            X_test_num_enc.columns = [i+'_binned' for i in X_test_num_enc.columns]
+
+            self.numerical_binned_ = disc.binner_dict_
+            # filling missing values on the results
+            X_train_num_enc, X_test_num_enc = X_train_num_enc.fillna(0), X_test_num_enc.fillna(0)
+
+
```

Comment:
For numerical variables, we should include both transformers into a pipeline, similar to what is done here:
https://github.com/solegalli/feature_engine/blob/master/feature_engine/encoding/decision_tree.py#L133